enhancements/cluster-logging/forward_to_cloudwatch.md

```diff
@@ -0,0 +1,309 @@
+  For example, when namespace re-creation is considered a security risk.
+
+### API fields
+
+New API fields in the `output.cloudwatch` section:
+
+- `region`: (string) AWS region name, required to connect.
+- `groupBy`: (string, default "category") Take group name from logging meta-data. Values:
+   - `category`: category of log entry - one of "application", "infrastructure", or "audit"
+   - `namespaceName`: Container's namespace name
+   - `namespaceUUID`: Container's namespace UUID
+
+Existing fields:
+
+- `url`: Not used in production. Sets the `endpoint` parameter in fluentd for use in testing.
+- `secret`: AWS credentials, the secret must contain keys `aws_access_key_id` and `aws_secret_access_key`.
+
+**Note**: The installer UI (Addon or OLM) can get AWS credentials from a `cloudcredential.openshift.io/v1`.
+The user only has to provide a `region` to enable CloudWatch forwarding for a cluster.
+Details are out of scope for this proposal.
+
+### User Stories
+
+**Note**: in all cases the CloudWatch *log stream names* are generated from the UUID of the source container or node as described above.
+The CloudWatch *log group names* are different depending on the use case.
+
+#### I want to forward logs to CloudWatch instead of a local store
+
+```
+apiVersion: "logging.openshift.io/v1"
+kind: "ClusterLogForwarder"
+spec:
+  outputs:
+  - name: CloudWatchOut
+	type: cloudwatch
+	cloudwatch:
+	  region: myregion
+	secret:
+	   name: mysecret
+  pipelines:
+  - inputRefs: [application, infrastructure, audit]
+    outputRefs: [CloudWatchOut]
+```
+
+CloudWatch group names are: "application", "infrastructure", "audit"
+
+#### I want to group application logs by namespace
+
+To group by namespace name:
+
+```
+apiVersion: "logging.openshift.io/v1"
+kind: "ClusterLogForwarder"
+spec:
+  outputs:
+  - name: CloudWatchOut
+	type: cloudwatch
+	cloudwatch:
+	  region: myregion
+	  groupBy: namespaceName
+	secret:
+	   name: mysecret
+  pipelines:
+  - inputRefs: [application, infrastructure, audit]
+    outputRefs: [CloudWatchOut]
+```
+
+CloudWatch group names are the names of all namespaces from which logs are collected.
+
+To group by namespace UUID instead, replace `namespaceName` with `namespaceUUID`.
+
+### Implementation Details
+
+Use the [fluentd CloudWatch plugin][plugin].
+Most of the CRD fields have an obvious corresponding fluentd parameter.
+
+- Set log group names using `log_group_name_key` to select the appropriate meta-data value
+- Set `auto_create_stream true` to create streams and groups on the fly.
+- Set `log-stream-name` to a a unique ID for the source of the log (see "CloudWatch streams and groups" above)
+
+For the first implementation, the log stream name will be:
+- *container logs*: use the fluentd tag, via the `use_tag_as_stream true` plugin setting. Unique because it includes container-id.
+- *node logs* (audit, infrastructure): use fluentd tag + node-id. Unique per node and per audit/infrastructure log file.
```

Comment:
I don't know what or how to get `node-id` but the node name is same as host name.  Landing on this which would give you journal and all audit logs grouped by node hostname:

```
<label {{.LabelName}}>
  <filter kubernetes.**>
    @type record_transformer
    <record>
      cw_group_name {{.LogGroupName }}
      cw_stream_name ${record["kubernetes"]["container_image_id"]}
    </record>
  </filter>
  <filter journal.** *audit.log>
    @type record_transformer
    <record>
      cw_group_name ${record["hostname"]}
      cw_stream_name ${tag}
    </record>
  </filter>
```